src/Cryptol/Eval/What4/Value.hs

```diff
@@ -100,7 +155,7 @@ instance W4.IsExprBuilder sym => Backend (What4 sym) where
 
   assertSideCondition _ cond err
     | Just False <- W4.asConstantPred cond = W4Eval (\_ -> pure (W4Error err))
```

Comment:
I'm concerned here that we might be throwing away important definitional equations in error cases.